apis/openstack-nova/src/main/java/org/jclouds/openstack/nova/v2_0/compute/NovaComputeServiceAdapter.java

```diff
@@ -123,41 +128,32 @@ public NovaComputeServiceAdapter(NovaApi novaApi, @Region Supplier<Set<String>>
          options.networks(templateOptions.getNetworks());
       }
 
-      Optional<String> privateKey = Optional.absent();
       if (templateOptions.getKeyPairName() != null) {
          options.keyPairName(templateOptions.getKeyPairName());
-         KeyPair keyPair = keyPairCache.getIfPresent(RegionAndName.fromRegionAndName(template.getLocation().getId(), templateOptions.getKeyPairName()));
-         if (keyPair != null && keyPair.getPrivateKey() != null) {
-            privateKey = Optional.of(keyPair.getPrivateKey());
-            credentialsBuilder.privateKey(privateKey.get());
-         }
```

Comment:
right! 
I've re-added same logic in `ApplyNovaTemplateOptionsCreateNodesWithGroupEncodedIntoNameThenAddToSet`
thanks!

Tested with
```
               templateOptions.generateKeyPair(true);
```
and with
```
               templateOptions.keyPairName("test");
               String privateKey = Files.toString(new File("/Users/andrea/Downloads/test.pem"), Charsets.UTF_8);
               templateOptions.overrideLoginPrivateKey(privateKey);
```
looks good to me